app/models/concerns/service_history/builder.rb

```diff
@@ -122,7 +122,11 @@ def clients_still_processing?(client_ids:)
     # Class method
     private def builder_batch_job_scope
       Delayed::Job.uncached do
-        Delayed::Job.where(failed_at: nil).jobs_for_class('ServiceHistory::RebuildEnrollments')
+        Delayed::Job.where(failed_at: nil).
+          # Cache buster - for whatever reason `uncached` is completely inconsistent.
+          # We ALWAYS want this to query the database.
+          where("1 != #{rand(2..50)}").
+          jobs_for_class('ServiceHistory::RebuildEnrollments')
```

Comment:
I wonder if this is because AR evaluation is lazy? So builder_batch_job_scope just returns the scope but it's the caller that executes it (which would be outside of the uncached block). I think that might be what's going on. Can we just return an array?
```suggestion
        Delayed::Job.where(failed_at: nil).
          jobs_for_class('ServiceHistory::RebuildEnrollments').
          to_a # force execution within uncached block
```

FWIW I think 2..50 is too small a space for cache busting, it seems like it would be unreliable as you're likely to generate the same random number repeatedly. You'd probably want a counter of some kind (`@counter += 1`) to avoid repeats